Client/Frontend/Browser/Playlist/Controllers/PlaylistFolderController.swift

```diff
@@ -0,0 +1,595 @@
+        deleteAction.backgroundColor = UIColor.braveErrorLabel
+        
+        return UISwipeActionsConfiguration(actions: [deleteAction])
+    }
+    
+    func tableView(_ tableView: UITableView, contextMenuConfigurationForRowAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        
+        guard let section = Section(rawValue: indexPath.section),
+              section == .folders else {
+            return nil
+        }
+        
+        guard let folder = othersFRC.fetchedObjects?[safe: indexPath.row],
+              let folderUUID = folder.uuid else {
+            return nil
+        }
+        
+        let actionProvider: UIContextMenuActionProvider = { [weak self] _ in
+            return UIMenu(children: [
+                UIAction(title: Strings.PlaylistFolders.playlistFolderEditButtonTitle, image: UIImage(systemName: "pencil"), handler: { _ in
+                    guard let self = self else {
+                        return
+                    }
+                    
+                    self.onEditFolder(folderUUID: folderUUID)
+                }),
+                
+                UIAction(title: Strings.delete, image: UIImage(systemName: "trash"), attributes: .destructive, handler: { _ in
+                    guard let self = self else { return }
+                    self.othersFRC.fetchedObjects?.first(where: { $0.uuid == folderUUID })?.do {
+                        self.deleteFolder(folder: $0)
+                    }
+                })
+            ])
+        }
+
+        let identifier = NSDictionary(dictionary: ["folderUUID": folderUUID,
+                                                   "row": indexPath.row])
+        return UIContextMenuConfiguration(identifier: identifier, previewProvider: nil, actionProvider: actionProvider)
+    }
+    
+    func tableView(_ tableView: UITableView, previewForHighlightingContextMenuWithConfiguration configuration: UIContextMenuConfiguration) -> UITargetedPreview? {
+        guard let identifier = configuration.identifier as? NSDictionary else {
+            return nil
+        }
+        
+        guard let folderUUID = identifier["folderUUID"] as? String,
+              let row = identifier["row"] as? Int,
+              row < othersFRC.fetchedObjects?.count ?? -1 else {
+            return nil
+        }
+        
+        guard let folder = othersFRC.fetchedObjects?[safe: row],
+              folderUUID == folder.uuid,
+              let cell = tableView.cellForRow(at: IndexPath(row: row, section: Section.folders.rawValue)) else {
+            return nil
+        }
+
+        let parameters = UIPreviewParameters()
+        parameters.visiblePath = UIBezierPath(roundedRect: cell.contentView.frame.with {
+            $0.size.width = cell.bounds.width
+        }, cornerRadius: 12.0)
+        parameters.backgroundColor = .tertiaryBraveBackground
+        
+        return UITargetedPreview(view: cell, parameters: parameters)
```

Comment:
When highlighting in table view, do we need to specify the TargetedPreview since we are already highlighting the table cell itself? Isnt it the default one ?

Reply:
No it is not default. I need to provide a preview for long-press. Otherwise, it does not render correctly.